{"nbformat":4,"nbformat_minor":0,"metadata":{"accelerator":"GPU","colab":{"name":"V1.ipynb","provenance":[],"collapsed_sections":[]},"kernelspec":{"display_name":"Python 3","name":"python3"}},"cells":[{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"Bf1J-Ia40djU","executionInfo":{"status":"ok","timestamp":1611667751832,"user_tz":-60,"elapsed":752,"user":{"displayName":"Gijs Mourits","photoUrl":"","userId":"15280450472625728178"}},"outputId":"cce3ac9e-653e-47f5-a6a3-5f58c9ede88a"},"source":["from google.colab import drive\r\n","drive.mount(\"/content/gdrive\")"],"execution_count":13,"outputs":[{"output_type":"stream","text":["Drive already mounted at /content/gdrive; to attempt to forcibly remount, call drive.mount(\"/content/gdrive\", force_remount=True).\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"3xX02JuT1pDd","executionInfo":{"status":"ok","timestamp":1611667779553,"user_tz":-60,"elapsed":470,"user":{"displayName":"Gijs Mourits","photoUrl":"","userId":"15280450472625728178"}},"outputId":"a38039cf-0acd-48bd-be1d-20f447e4b326"},"source":["%cd /content/gdrive/My Drive/Colorization_Colab/Colorization-PR-V1"],"execution_count":16,"outputs":[{"output_type":"stream","text":["/content/gdrive/My Drive/Colorization_Colab/Colorization-PR-V1\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"cmwSyFQW17Pf"},"source":["#! git clone https://github.com/gimoAI/Colorization-PR-V1.git"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"sKHJOFOY2vbA"},"source":["#! git pull"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"VHv0HdDQ2yPi"},"source":["!pip install numpy\r\n","!pip install scipy\r\n","!pip install future\r\n","!pip install matplotlib\r\n","!pip install pillow\r\n","!pip install tensorflow==2.4\r\n","!pip install keras"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"QqCx1YmVoELr"},"source":[""],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"riPaIe0H4n0T","executionInfo":{"status":"ok","timestamp":1611667803667,"user_tz":-60,"elapsed":11150,"user":{"displayName":"Gijs Mourits","photoUrl":"","userId":"15280450472625728178"}},"outputId":"d81153c7-a5e4-42b3-d3fe-d75c1308f942"},"source":["!python test.py --checkpoints-path ./checkpoints --test-input ./checkpoints/test --test-output ./checkpoints/output"],"execution_count":18,"outputs":[{"output_type":"stream","text":["2021-01-26 13:29:53.449920: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.10.1\n","WARNING:tensorflow:From /usr/local/lib/python3.6/dist-packages/tensorflow/python/compat/v2_compat.py:96: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.\n","Instructions for updating:\n","non-resource variables are not supported in the long term\n","2021-01-26 13:29:55.477524: I tensorflow/compiler/jit/xla_cpu_device.cc:41] Not creating XLA devices, tf_xla_enable_xla_devices not set\n","2021-01-26 13:29:55.478417: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcuda.so.1\n","2021-01-26 13:29:55.514559: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:941] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2021-01-26 13:29:55.515261: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1720] Found device 0 with properties: \n","pciBusID: 0000:00:04.0 name: Tesla T4 computeCapability: 7.5\n","coreClock: 1.59GHz coreCount: 40 deviceMemorySize: 14.73GiB deviceMemoryBandwidth: 298.08GiB/s\n","2021-01-26 13:29:55.515308: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.10.1\n","2021-01-26 13:29:55.517325: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcublas.so.10\n","2021-01-26 13:29:55.517385: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcublasLt.so.10\n","2021-01-26 13:29:55.519385: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcufft.so.10\n","2021-01-26 13:29:55.519854: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcurand.so.10\n","2021-01-26 13:29:55.522020: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcusolver.so.10\n","2021-01-26 13:29:55.522970: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcusparse.so.10\n","2021-01-26 13:29:55.526998: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudnn.so.7\n","2021-01-26 13:29:55.527111: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:941] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2021-01-26 13:29:55.527663: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:941] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2021-01-26 13:29:55.528168: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1862] Adding visible gpu devices: 0\n","2021-01-26 13:29:55.528590: I tensorflow/compiler/jit/xla_gpu_device.cc:99] Not creating XLA devices, tf_xla_enable_xla_devices not set\n","2021-01-26 13:29:55.528721: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:941] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2021-01-26 13:29:55.529240: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1720] Found device 0 with properties: \n","pciBusID: 0000:00:04.0 name: Tesla T4 computeCapability: 7.5\n","coreClock: 1.59GHz coreCount: 40 deviceMemorySize: 14.73GiB deviceMemoryBandwidth: 298.08GiB/s\n","2021-01-26 13:29:55.529289: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.10.1\n","2021-01-26 13:29:55.529320: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcublas.so.10\n","2021-01-26 13:29:55.529347: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcublasLt.so.10\n","2021-01-26 13:29:55.529370: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcufft.so.10\n","2021-01-26 13:29:55.529390: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcurand.so.10\n","2021-01-26 13:29:55.529411: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcusolver.so.10\n","2021-01-26 13:29:55.529432: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcusparse.so.10\n","2021-01-26 13:29:55.529454: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudnn.so.7\n","2021-01-26 13:29:55.529516: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:941] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2021-01-26 13:29:55.530060: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:941] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2021-01-26 13:29:55.530538: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1862] Adding visible gpu devices: 0\n","2021-01-26 13:29:55.530580: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.10.1\n","2021-01-26 13:29:56.229609: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1261] Device interconnect StreamExecutor with strength 1 edge matrix:\n","2021-01-26 13:29:56.229671: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1267]      0 \n","2021-01-26 13:29:56.229688: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1280] 0:   N \n","2021-01-26 13:29:56.229881: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:941] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2021-01-26 13:29:56.230461: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:941] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2021-01-26 13:29:56.231014: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:941] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2021-01-26 13:29:56.231489: W tensorflow/core/common_runtime/gpu/gpu_bfc_allocator.cc:39] Overriding allow_growth setting because the TF_FORCE_GPU_ALLOW_GROWTH environment variable is set. Original config value was 0.\n","2021-01-26 13:29:56.231536: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1406] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 13960 MB memory) -> physical GPU (device: 0, name: Tesla T4, pci bus id: 0000:00:04.0, compute capability: 7.5)\n","/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/legacy_tf_layers/convolutional.py:414: UserWarning: `tf.layers.conv2d` is deprecated and will be removed in a future version. Please Use `tf.keras.layers.Conv2D` instead.\n","  warnings.warn('`tf.layers.conv2d` is deprecated and '\n","/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/engine/base_layer_v1.py:1719: UserWarning: `layer.apply` is deprecated and will be removed in a future version. Please use `layer.__call__` method instead.\n","  warnings.warn('`layer.apply` is deprecated and '\n","/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/legacy_tf_layers/normalization.py:308: UserWarning: `tf.layers.batch_normalization` is deprecated and will be removed in a future version. Please use `tf.keras.layers.BatchNormalization` instead. In particular, `tf.control_dependencies(tf.GraphKeys.UPDATE_OPS)` should not be used (consult the `tf.keras.layers.BatchNormalization` documentation).\n","  '`tf.layers.batch_normalization` is deprecated and '\n","/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/legacy_tf_layers/convolutional.py:1294: UserWarning: `tf.layers.conv2d_transpose` is deprecated and will be removed in a future version. Please Use `tf.keras.layers.Conv2DTranspose` instead.\n","  warnings.warn('`tf.layers.conv2d_transpose` is deprecated and '\n","2021-01-26 13:29:58.535454: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:196] None of the MLIR optimization passes are enabled (registered 0 passes)\n","2021-01-26 13:29:58.565457: I tensorflow/core/platform/profile_utils/cpu_utils.cc:112] CPU Frequency: 2199995000 Hz\n","\n","Testing...\n","2021-01-26 13:30:00.332798: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudnn.so.7\n","2021-01-26 13:30:01.867159: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcublas.so.10\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py\", line 1375, in _do_call\n","    return fn(*args)\n","  File \"/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py\", line 1360, in _run_fn\n","    target_list, run_metadata)\n","  File \"/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py\", line 1453, in _call_tf_sessionrun\n","    run_metadata)\n","tensorflow.python.framework.errors_impl.InvalidArgumentError: 2 root error(s) found.\n","  (0) Invalid argument: ConcatOp : Dimensions of inputs should match: shape[0] = [1,512,1,1] vs. shape[1] = [1,512,2,2]\n","\t [[{{node gen_1/concat}}]]\n","\t [[output/_9]]\n","  (1) Invalid argument: ConcatOp : Dimensions of inputs should match: shape[0] = [1,512,1,1] vs. shape[1] = [1,512,2,2]\n","\t [[{{node gen_1/concat}}]]\n","0 successful operations.\n","0 derived errors ignored.\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"test.py\", line 5, in <module>\n","    main(options)\n","  File \"/content/gdrive/My Drive/Colorization_Colab/Colorization-PR-V1/src/main.py\", line 60, in main\n","    model.test()\n","  File \"/content/gdrive/My Drive/Colorization_Colab/Colorization-PR-V1/src/models.py\", line 129, in test\n","    outputs = self.sess.run(self.sampler, feed_dict=feed_dic)\n","  File \"/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py\", line 968, in run\n","    run_metadata_ptr)\n","  File \"/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py\", line 1191, in _run\n","    feed_dict_tensor, options, run_metadata)\n","  File \"/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py\", line 1369, in _do_run\n","    run_metadata)\n","  File \"/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py\", line 1394, in _do_call\n","    raise type(e)(node_def, op, message)\n","tensorflow.python.framework.errors_impl.InvalidArgumentError: 2 root error(s) found.\n","  (0) Invalid argument: ConcatOp : Dimensions of inputs should match: shape[0] = [1,512,1,1] vs. shape[1] = [1,512,2,2]\n","\t [[node gen_1/concat (defined at /content/gdrive/My Drive/Colorization_Colab/Colorization-PR-V1/src/networks.py:109) ]]\n","\t [[output/_9]]\n","  (1) Invalid argument: ConcatOp : Dimensions of inputs should match: shape[0] = [1,512,1,1] vs. shape[1] = [1,512,2,2]\n","\t [[node gen_1/concat (defined at /content/gdrive/My Drive/Colorization_Colab/Colorization-PR-V1/src/networks.py:109) ]]\n","0 successful operations.\n","0 derived errors ignored.\n","\n","Errors may have originated from an input operation.\n","Input Source operations connected to node gen_1/concat:\n"," gen_1/Relu (defined at /content/gdrive/My Drive/Colorization_Colab/Colorization-PR-V1/src/ops.py:52)\t\n"," gen_1/LeakyRelu_6 (defined at /content/gdrive/My Drive/Colorization_Colab/Colorization-PR-V1/src/ops.py:28)\n","\n","Input Source operations connected to node gen_1/concat:\n"," gen_1/Relu (defined at /content/gdrive/My Drive/Colorization_Colab/Colorization-PR-V1/src/ops.py:52)\t\n"," gen_1/LeakyRelu_6 (defined at /content/gdrive/My Drive/Colorization_Colab/Colorization-PR-V1/src/ops.py:28)\n","\n","Original stack trace for 'gen_1/concat':\n","  File \"test.py\", line 5, in <module>\n","    main(options)\n","  File \"/content/gdrive/My Drive/Colorization_Colab/Colorization-PR-V1/src/main.py\", line 40, in main\n","    model.build()\n","  File \"/content/gdrive/My Drive/Colorization_Colab/Colorization-PR-V1/src/models.py\", line 213, in build\n","    self.sampler = tf.identity(gen_factory.create(self.input_gray, kernel, seed, reuse_variables=True), name='output')\n","  File \"/content/gdrive/My Drive/Colorization_Colab/Colorization-PR-V1/src/networks.py\", line 109, in create\n","    output = tf.concat([layers[len(layers) - index - 2], output], axis=3)\n","  File \"/usr/local/lib/python3.6/dist-packages/tensorflow/python/util/dispatch.py\", line 201, in wrapper\n","    return target(*args, **kwargs)\n","  File \"/usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/array_ops.py\", line 1677, in concat\n","    return gen_array_ops.concat_v2(values=values, axis=axis, name=name)\n","  File \"/usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/gen_array_ops.py\", line 1208, in concat_v2\n","    \"ConcatV2\", values=values, axis=axis, name=name)\n","  File \"/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/op_def_library.py\", line 750, in _apply_op_helper\n","    attrs=attr_protos, op_def=op_def)\n","  File \"/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/ops.py\", line 3536, in _create_op_internal\n","    op_def=op_def)\n","  File \"/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/ops.py\", line 1990, in __init__\n","    self._traceback = tf_stack.extract_stack()\n","\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"EvJjf768oElP"},"source":[""]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"KpxkKN4-_TiQ","executionInfo":{"status":"ok","timestamp":1611666610498,"user_tz":-60,"elapsed":358524,"user":{"displayName":"Gijs Mourits","photoUrl":"","userId":"15280450472625728178"}},"outputId":"874d6aa7-7e63-4f28-c7e3-d085c056488a"},"source":["!python train.py --seed 100 --dataset cifar10 --dataset-path ./dataset/cifar10 --checkpoints-path ./checkpoints --gpu-ids '0' --save-interval 500 --batch-size 128 --epochs 1  --lr 3e-4 --lr-decay-steps 1e4 --augment True --acc-thresh 2.0"],"execution_count":null,"outputs":[{"output_type":"stream","text":["2021-01-26 13:04:12.346001: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.10.1\n","WARNING:tensorflow:From /usr/local/lib/python3.6/dist-packages/tensorflow/python/compat/v2_compat.py:96: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.\n","Instructions for updating:\n","non-resource variables are not supported in the long term\n","2021-01-26 13:04:14.401388: I tensorflow/compiler/jit/xla_cpu_device.cc:41] Not creating XLA devices, tf_xla_enable_xla_devices not set\n","2021-01-26 13:04:14.402257: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcuda.so.1\n","2021-01-26 13:04:14.438447: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:941] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2021-01-26 13:04:14.439060: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1720] Found device 0 with properties: \n","pciBusID: 0000:00:04.0 name: Tesla T4 computeCapability: 7.5\n","coreClock: 1.59GHz coreCount: 40 deviceMemorySize: 14.73GiB deviceMemoryBandwidth: 298.08GiB/s\n","2021-01-26 13:04:14.439104: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.10.1\n","2021-01-26 13:04:14.440713: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcublas.so.10\n","2021-01-26 13:04:14.440785: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcublasLt.so.10\n","2021-01-26 13:04:14.442505: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcufft.so.10\n","2021-01-26 13:04:14.442857: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcurand.so.10\n","2021-01-26 13:04:14.444752: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcusolver.so.10\n","2021-01-26 13:04:14.445657: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcusparse.so.10\n","2021-01-26 13:04:14.449328: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudnn.so.7\n","2021-01-26 13:04:14.449446: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:941] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2021-01-26 13:04:14.450036: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:941] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2021-01-26 13:04:14.450739: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1862] Adding visible gpu devices: 0\n","2021-01-26 13:04:14.451198: I tensorflow/compiler/jit/xla_gpu_device.cc:99] Not creating XLA devices, tf_xla_enable_xla_devices not set\n","2021-01-26 13:04:14.451312: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:941] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2021-01-26 13:04:14.451846: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1720] Found device 0 with properties: \n","pciBusID: 0000:00:04.0 name: Tesla T4 computeCapability: 7.5\n","coreClock: 1.59GHz coreCount: 40 deviceMemorySize: 14.73GiB deviceMemoryBandwidth: 298.08GiB/s\n","2021-01-26 13:04:14.451883: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.10.1\n","2021-01-26 13:04:14.451922: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcublas.so.10\n","2021-01-26 13:04:14.451944: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcublasLt.so.10\n","2021-01-26 13:04:14.451965: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcufft.so.10\n","2021-01-26 13:04:14.451986: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcurand.so.10\n","2021-01-26 13:04:14.452004: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcusolver.so.10\n","2021-01-26 13:04:14.452023: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcusparse.so.10\n","2021-01-26 13:04:14.452041: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudnn.so.7\n","2021-01-26 13:04:14.452109: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:941] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2021-01-26 13:04:14.452633: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:941] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2021-01-26 13:04:14.453135: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1862] Adding visible gpu devices: 0\n","2021-01-26 13:04:14.453178: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.10.1\n","2021-01-26 13:04:15.160168: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1261] Device interconnect StreamExecutor with strength 1 edge matrix:\n","2021-01-26 13:04:15.160223: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1267]      0 \n","2021-01-26 13:04:15.160246: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1280] 0:   N \n","2021-01-26 13:04:15.160443: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:941] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2021-01-26 13:04:15.161343: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:941] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2021-01-26 13:04:15.162217: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:941] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2021-01-26 13:04:15.162835: W tensorflow/core/common_runtime/gpu/gpu_bfc_allocator.cc:39] Overriding allow_growth setting because the TF_FORCE_GPU_ALLOW_GROWTH environment variable is set. Original config value was 0.\n","2021-01-26 13:04:15.162905: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1406] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 13960 MB memory) -> physical GPU (device: 0, name: Tesla T4, pci bus id: 0000:00:04.0, compute capability: 7.5)\n","/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/legacy_tf_layers/convolutional.py:414: UserWarning: `tf.layers.conv2d` is deprecated and will be removed in a future version. Please Use `tf.keras.layers.Conv2D` instead.\n","  warnings.warn('`tf.layers.conv2d` is deprecated and '\n","/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/engine/base_layer_v1.py:1719: UserWarning: `layer.apply` is deprecated and will be removed in a future version. Please use `layer.__call__` method instead.\n","  warnings.warn('`layer.apply` is deprecated and '\n","/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/legacy_tf_layers/normalization.py:308: UserWarning: `tf.layers.batch_normalization` is deprecated and will be removed in a future version. Please use `tf.keras.layers.BatchNormalization` instead. In particular, `tf.control_dependencies(tf.GraphKeys.UPDATE_OPS)` should not be used (consult the `tf.keras.layers.BatchNormalization` documentation).\n","  '`tf.layers.batch_normalization` is deprecated and '\n","/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/legacy_tf_layers/convolutional.py:1294: UserWarning: `tf.layers.conv2d_transpose` is deprecated and will be removed in a future version. Please Use `tf.keras.layers.Conv2DTranspose` instead.\n","  warnings.warn('`tf.layers.conv2d_transpose` is deprecated and '\n","WARNING:tensorflow:From /usr/local/lib/python3.6/dist-packages/tensorflow/python/util/dispatch.py:201: calling dropout (from tensorflow.python.ops.nn_ops) with keep_prob is deprecated and will be removed in a future version.\n","Instructions for updating:\n","Please use `rate` instead of `keep_prob`. Rate should be set to `rate = 1 - keep_prob`.\n","2021-01-26 13:04:16.885539: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:196] None of the MLIR optimization passes are enabled (registered 0 passes)\n","2021-01-26 13:04:16.907801: I tensorflow/core/platform/profile_utils/cpu_utils.cc:112] CPU Frequency: 2199995000 Hz\n","loading model...\n","\n","\n","------------ Options -------------\n","acc_thresh: 2.0\n","augment: True\n","batch_size: 128\n","beta1: 0\n","checkpoints_path: ./checkpoints/cifar10\n","color_space: LAB\n","dataset: cifar10\n","dataset_path: ./dataset/cifar10\n","epochs: 1\n","gpu_ids: 0\n","l1_weight: 100.0\n","label_smoothing: False\n","log: False\n","log_interval: 10\n","lr: 0.0003\n","lr_decay: True\n","lr_decay_rate: 0.1\n","lr_decay_steps: 10000.0\n","mode: 0\n","name: CGAN\n","sample: True\n","sample_interval: 1000\n","sample_size: 8\n","save: True\n","save_interval: 500\n","seed: 100\n","test_input: \n","test_output: \n","training: False\n","turing_test_delay: 0\n","turing_test_size: 100\n","validate: True\n","validate_interval: 0\n","visualize: False\n","visualize_window: 100\n","-------------- End ----------------\n","\n","Training epoch: 1 - learning rate: 1e-06\n","2021-01-26 13:04:18.773007: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcublas.so.10\n","2021-01-26 13:04:19.027897: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudnn.so.7\n","49438/50000 [=======================>.] - ETA: 3s - epoch: 1 - iter: 391 - step: 76391 - D loss: 1.3863 - D fake: 0.6932 - D real: 0.6932 - G loss: 3.3242 - G L1: 2.6310 - G gan: 0.6932 - accuracy: 0.5766StopIteration prevented, what happens now?\n","\n","\n","Validating epoch: 1\n"," 9894/10000 [=======================>.] - ETA: 0s - D loss: 1.3863 - D fake: 0.6932 - D real: 0.6931 - G loss: 7.1046 - G L1: 6.4115 - G gan: 0.6931 - accuracy: 0.1949StopIteration prevented, what happens now?\n","\n","\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"jpi4eCUbKObt"},"source":["while True:pass\r\n","#Run this while running the above"],"execution_count":null,"outputs":[]}]}